src/analysis/calculus/fderiv.lean

```diff
@@ -2867,4 +2867,36 @@ lemma differentiable.restrict_scalars (h : differentiable 𝕜' f) :
   differentiable 𝕜 f :=
 λx, (h x).restrict_scalars 𝕜
 
+lemma has_fderiv_within_at_of_eq {s : set E} {g' : E →L[𝕜] F} (h : has_fderiv_within_at f g' s x)
+  (H : f'.restrict_scalars 𝕜 = g') : has_fderiv_within_at f f' s x :=
+by { simp only [has_fderiv_within_at, has_fderiv_at_filter] at h ⊢,
+     rwa [← f'.coe_restrict_scalars', H] }
+
+lemma has_fderiv_at_of_eq {g' : E →L[𝕜] F} (h : has_fderiv_at f g' x)
+  (H : f'.restrict_scalars 𝕜 = g') : has_fderiv_at f f' x :=
+by simp only [has_fderiv_at, has_fderiv_at_filter] at h ⊢; rwa [← f'.coe_restrict_scalars', H]
+
+lemma fderiv_eq_fderiv (h : differentiable_at 𝕜' f x) :
+  (fderiv 𝕜 f x : E → F) = fderiv 𝕜' f x :=
+by rw [(h.restrict_scalars 𝕜).has_fderiv_at.unique (h.has_fderiv_at.restrict_scalars 𝕜),
+       coe_restrict_scalars']
```

Comment:
More powerful to state that they agree as `𝕜`-continuous-linear-maps rather than just as functions:
```suggestion
  fderiv 𝕜 f x = (fderiv 𝕜' f x).restrict_scalars 𝕜 :=
(h.has_fderiv_at.restrict_scalars 𝕜).fderiv
```
Also, can you please rename this to allow for projection notation?  (Maybe `differentiable_at.fderiv_restrict_scalars`.)